pkg/kudoctl/cmd/upgrade.go

```diff
@@ -0,0 +1,165 @@
+	}
+	if options.InstanceName == "" {
+		return fmt.Errorf("please use --instance and specify instance name. It cannot be empty")
+	}
+
+	return nil
+}
+
+func runUpgrade(args []string, options *options) error {
+	err := validate(args, options)
+	if err != nil {
+		return err
+	}
+	packageToUpgrade := args[0]
+
+	kc, err := kudo.NewClient(options.Namespace, viper.GetString("kubeconfig"))
+	if err != nil {
+		return errors.Wrap(err, "creating kudo client")
+	}
+
+	// Resolve the package to upgrade to
+	repository, err := repo.NewOperatorRepository(repo.Default)
+	if err != nil {
+		return errors.WithMessage(err, "could not build operator repository")
+	}
+	crds, err := install.GetPackageCRDs(packageToUpgrade, options.PackageVersion, repository)
+	if err != nil {
+		return errors.Wrapf(err, "failed to resolve package CRDs for operator: %s", packageToUpgrade)
+	}
+
+	return upgrade(crds.OperatorVersion, kc, options)
+}
+
+func upgrade(newOv *v1alpha1.OperatorVersion, kc *kudo.Client, options *options) error {
+	operatorName := newOv.Spec.Operator.Name
+	nextOperatorVersion := newOv.Spec.Version
+
+	// Make sure the instance you want to upgrade exists
+	instance, err := kc.GetInstance(options.InstanceName, options.Namespace)
+	if err != nil {
+		return errors.Wrapf(err, "verifying the instance does not already exist")
+	}
+	if instance == nil {
+		return fmt.Errorf("instance %s in namespace %s does not exist in the cluster", options.InstanceName, options.Namespace)
+	}
+
+	// Check OperatorVersion and if upgraded version is higher than current version
+	ov, err := kc.GetOperatorVersion(instance.Spec.OperatorVersion.Name, options.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "retrieving existing operator version")
+	}
+	if ov == nil {
+		return fmt.Errorf("no operator version for this operator installed yet for %s in namespace %s. Please use install command if you want to install new operator into cluster", operatorName, options.Namespace)
+	}
+	oldVersion, err := semver.NewVersion(ov.Spec.Version)
+	if err != nil {
+		return errors.Wrapf(err, "when parsing %s as semver", ov.Spec.Version)
+	}
+	newVersion, err := semver.NewVersion(nextOperatorVersion)
+	if err != nil {
+		return errors.Wrapf(err, "when parsing %s as semver", nextOperatorVersion)
+	}
+	if !oldVersion.LessThan(newVersion) {
+		return fmt.Errorf("upgraded version %s is the same or smaller as current version %s -> not upgrading", nextOperatorVersion, ov.Spec.Version)
+	}
+
+	// install OV
+	versionsInstalled, err := kc.OperatorVersionsInstalled(operatorName, options.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "retrieving existing operator versions")
+	}
+	if !install.VersionExists(versionsInstalled, nextOperatorVersion) {
+		if _, err := kc.InstallOperatorVersionObjToCluster(newOv, options.Namespace); err != nil {
+			return errors.Wrapf(err, "failed installing OperatorVersion for operator: %s", operatorName)
```

Comment:
the error should include the `newOv` with the oName

Reply:
I put just the version there, I don't like printing out huuuge yaml into cli output by default